### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Disability Benefits Experience Team 2, Disability Comp Application (21-526 EZ), Updated 0781 Flow

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#94389]
## Next Steps for the VFS team

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Very interested to see the form-within-a-form concept take shape!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Consider:** Looking elsewhere on VA.gov, the [supplemental claims flow](https://www.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/start) also features a conditional in which an additional form is filed on behalf of the Veteran. That flow doesn't make the fact that you're doing a form-within-a-form as explicit, but it does insert an additional VA form number into the header of the form. I'm not recommending that you match them precisely, but I do encourage you to look at that implementation and see if there's any inspiration you can draw from there / any consistency of experience you can build between the two forms.
- [x] **Must:** Be careful about your use of links and buttons, since these have some differences in how certain groups of assistive technology users interact with them. The general rule is that if it navigates it's a link, if it does something (eg. saving user data) it's a button.
   
   Two spots stand out to me:
   - The ["Opt out of this form" element](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=8423-101970&t=ZPizI1OLetQ0Qs2J-4) opens a modal and eventually results in a user preference being saved, so this should be a button. I think some variation on the [button icon component](https://design.va.gov/components/button/button-icon) would be appropriate without overwhelming the "start this form" link visually, but note that the component doesn't support custom text out of the box.
   - The ["Add an event" element](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=8742-33305&t=ZPizI1OLetQ0Qs2J-4) just moves users to a new page (similar to the "start this form" link), so this should be a link.
- [x] **Coding note:** As far as I know, most (all?) of the form input components in the design system take only text strings for input labels and hint text --- which means no text styling and no line breaks in those spots. Looking at ["Details about your traumatic event"](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=8742-33159&t=ZPizI1OLetQ0Qs2J-4) and the [va-textarea component](https://design.va.gov/storybook/?path=/docs/uswds-va-textarea--docs) as an example, you may need to rearrange some things and/or tweak your design to fit how the components are structured.
- [x] **Coding note:** When a screen reader user navigates through a form flow, they will typically try to move through the form as efficiently as they can (like anyone else!), which often means pressing [tab] to jump through the interactive elements rather than having their screen reader announce everything on each page. That means that helper text placed before, between, and after inputs can be missed by screen reader users unless it's programmatically associated with the inputs.
   
   There is a LOT of text on some of these pages. Some of it is definitely critical to understanding and answering the questions and definitely needs to be programmatically associated, and some of it may not be as necessary and can be skipped. You know your product best and can make that determination, but I'd definitely encourage you to define "critical to understanding and answering" broadly enough to include the trauma-informed language that's used throughout. I think you do a great job of providing context and gentle directions around some tough topics and I wouldn't want that to be lost.
   
   So with that preamble out of the way, a good approach for programmatically associating a ton of helper text is to put it into the `legend` and then styling it to look however you want it to look. It's kind of clunky HTML but it's more reliable than some ARIA techniques. Example:
   ```
   <fieldset>
      <legend>
         <h3 class="some-heading-class">Heading [will be announced when tabbing into the first input in the fieldset]</h3>
         <p class="some-helper-text-class">Helper text, inside the legend but styled to look like regular text [will be announced when tabbing into the first input in the fieldset]</p>
      </legend>
      <p>Helper text, outside the legend [will NOT be announced when tabbing into the first input in the fieldset]</p>
      <va-text-input label="Input label [will be announced etc]" hint="Hint text [will be announced etc]" ... />
   </fieldset>
   ```

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone

**Comment by jinaryu** [comment]
`Consider: Looking elsewhere on VA.gov, the [supplemental claims flow](https://www.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/start) also features a conditional in which an additional form is filed on behalf of the Veteran. That flow doesn't make the fact that you're doing a form-within-a-form as explicit, but it does insert an additional VA form number into the header of the form. I'm not recommending that you match them precisely, but I do encourage you to look at that implementation and see if there's any inspiration you can draw from there / any consistency of experience you can build between the two forms.`

Responding to this! We have synced with the supp.claims team about the headers of forms within a form. They liked our approach and were interested to see how it tests. We have also synced about our evidence sections and did what we could do make things consistent between the two experiences

**Comment by jinaryu** [comment]
`Must: Be careful about your use of links and buttons, since these have some differences in how certain groups of assistive technology users interact with them. The general rule is that if it navigates it's a link, if it does something (eg. saving user data) it's a button.`

Changed opt out to button - icon
![Image](https://github.com/user-attachments/assets/6b721714-6772-45ec-965d-26fa101c65a6)


Changed add event to link
![Image](https://github.com/user-attachments/assets/e37dd68f-30ef-4689-8a5b-4ad706ffe69f)

**Comment by jinaryu** [comment]
`Coding note: As far as I know, most (all?) of the form input components in the design system take only text strings for input labels and hint text --- which means no text styling and no line breaks in those spots. Looking at ["Details about your traumatic event"](https://www.figma.com/design/r3Aj9FtLFS989mlVeBsgJg/0781-Redesign?node-id=8742-33159&t=ZPizI1OLetQ0Qs2J-4) and the [va-textarea component](https://design.va.gov/storybook/?path=/docs/uswds-va-textarea--docs) as an example, you may need to rearrange some things and/or tweak your design to fit how the components are structured.`

Removed styling from hint text ![Image](https://github.com/user-attachments/assets/883aabf4-3da3-4449-9e82-6d054f641cfd)

**Comment by jinaryu** [comment]
Thanks for the coding notes Brian! I've added these to our Figma for our devs to reference. Closing this ticket because all comments have been addressed
